### PR TITLE
[Improvement] Add List Schools Command into DG, edit NFR, add on user requirement details

### LINKED ISSUE #65: Document List Schools design and implementation into DG

### PR DESCRIPTION
**Pull request by louisjoety**
Fixes #65

### CODE REVIEW

File: uml-images/ListSchoolsCommand.png
Comment (chewycj) [inline uml-images/ListSchoolsCommand.png:1]: I like the neatness of the sequence diagram, it gives a clear overview of how the command works!

I would like to give some suggestions:
1. For the self-call, I think would be great to include the dashed line to show the context switch from displaySchoolList back to the execute method, after the list has been printed! 
2. For the entities (ListSchoolCommand, Command, and JsonObject), should there be a colon in front of it? Such as :ListSchoolCommand, :Command, and :JsonObject?

**Review by chewycj — APPROVED** [review]
LGTM!! There are some minor suggestions for the UML diagram, but overall it looks great and the published view of the DG looks neat and clear! 💯

File: uml-images/ListSchoolsCommand.png
Comment (louisjoety) [reply uml-images/ListSchoolsCommand.png:1]: image

I think no need for point 2, the `count` here doesn't seem to have a type? @chewycj

Comment (louisjoety) [reply uml-images/ListSchoolsCommand.png:1]: Okay, I made the changes already and simplified the diagram also, what do you think? 😸 @chewycj

Comment (wahkia) [reply uml-images/ListSchoolsCommand.png:1]: Nice sequence diagram! I would like to corroborate point 2 made by @chewycj. I observed prof declaring the type of the arguments in his lecture. A specific example is shown below:

image

Comment (louisjoety) [reply uml-images/ListSchoolsCommand.png:1]: good catch, i'll fix it, thanks!👍 @wahkia

Comment (chewycj) [reply uml-images/ListSchoolsCommand.png:1]: The UML looks great! 🥇

Comment (louisjoety) [reply uml-images/ListSchoolsCommand.png:1]: Will merge, I will continue to refine and refactor in subsequent DG PRs accordingly.

**Pull request merged**